### PR TITLE
ENVIRONMENT_DEPENDENCIES are now separated by comma

### PR DESCRIPTION
**Pull request by spisiakm**
#### Short description of what this resolves:

https://github.com/arquillian/arquillian-cube/issues/931

#### Changes proposed in this pull request:

- use a comma (and arbitrarily long string of whitespace chars) instead of just whitespaces to separate multiple values specified in `arquillian.xml` configuration file

### CODE REVIEW

**Comment by spisiakm** [comment]
I also noticed that multiple values in `WAIT_FOR_SERVICE_LIST` property are also separated by just whitespaces. That probably needs to be changed as well.

**Comment by lordofthejars** [comment]
Good, thank you very much, can you provide the fix for `WAIT_FOR_SERVICE_LIST ` and update the docs?

Thank you very much.

**Comment by cescoffier** [comment]
👍

File: kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/DefaultConfiguration.java [hunk 1]
```diff
@@ -132,7 +132,7 @@ public static DefaultConfiguration fromMap(Map<String, String> map) {
                     asUrlOrResource(getStringProperty(ENVIRONMENT_TEARDOWN_SCRIPT_URL, map, null)))
                 .withEnvironmentConfigUrl(getKubernetesConfigurationUrl(map))
                 .withEnvironmentDependencies(
-                    asURL(Strings.splitAndTrimAsList(getStringProperty(ENVIRONMENT_DEPENDENCIES, map, ""), "\\s+")))
+                    asURL(Strings.splitAndTrimAsList(getStringProperty(ENVIRONMENT_DEPENDENCIES, map, ""), ",\\s*")))
```
Comment (dipak-pawar) [inline kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/DefaultConfiguration.java]: How about `\\s*,\\s*` regex?

File: kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/DefaultConfiguration.java [hunk 2]
```diff
@@ -151,7 +151,7 @@ public static DefaultConfiguration fromMap(Map<String, String> map) {
                 .withWaitTimeout(getLongProperty(WAIT_TIMEOUT, map, DEFAULT_WAIT_TIMEOUT))
                 .withWaitPollInterval(getLongProperty(WAIT_POLL_INTERVAL, map, DEFAULT_WAIT_POLL_INTERVAL))
                 .withWaitForServiceList(
-                    Strings.splitAndTrimAsList(getStringProperty(WAIT_FOR_SERVICE_LIST, map, ""), "\\s+"))
+                    Strings.splitAndTrimAsList(getStringProperty(WAIT_FOR_SERVICE_LIST, map, ""), ",\\s*"))
```
Comment (dipak-pawar) [inline kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/DefaultConfiguration.java]: How about `\\s*,\\s*` regex?

File: openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/client/CubeOpenShiftConfiguration.java [hunk 3]
```diff
@@ -164,7 +164,7 @@ public static CubeOpenShiftConfiguration fromMap(Map<String, String> map) {
                     asUrlOrResource(getStringProperty(ENVIRONMENT_TEARDOWN_SCRIPT_URL, map, null)))
                 .withEnvironmentConfigUrl(getKubernetesConfigurationUrl(map))
                 .withEnvironmentDependencies(
-                    asURL(Strings.splitAndTrimAsList(getStringProperty(ENVIRONMENT_DEPENDENCIES, map, ""), "\\s+")))
+                    asURL(Strings.splitAndTrimAsList(getStringProperty(ENVIRONMENT_DEPENDENCIES, map, ""), ",\\s*")))
```
Comment (dipak-pawar) [inline openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/client/CubeOpenShiftConfiguration.java]: same as above

File: openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/client/CubeOpenShiftConfiguration.java [hunk 4]
```diff
@@ -183,7 +183,7 @@ public static CubeOpenShiftConfiguration fromMap(Map<String, String> map) {
                 .withWaitTimeout(getLongProperty(WAIT_TIMEOUT, map, DEFAULT_WAIT_TIMEOUT))
                 .withWaitPollInterval(getLongProperty(WAIT_POLL_INTERVAL, map, DEFAULT_WAIT_POLL_INTERVAL))
                 .withWaitForServiceList(
-                    Strings.splitAndTrimAsList(getStringProperty(WAIT_FOR_SERVICE_LIST, map, ""), "\\s+"))
+                    Strings.splitAndTrimAsList(getStringProperty(WAIT_FOR_SERVICE_LIST, map, ""), ",\\s*"))
```
Comment (dipak-pawar) [inline openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/client/CubeOpenShiftConfiguration.java]: same as above

File: kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/DefaultConfiguration.java (hunk 1 above)
Comment (spisiakm) [reply kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/DefaultConfiguration.java]: I don't mind, let me update it :)

Comment (lordofthejars) [reply kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/DefaultConfiguration.java]: I am not sure about any of both expressions, I think that the comma part should be optional, so the property can be either `mykubernetes.json` or `mykubernetes.json, mysecondkubernetes.json`

Comment (dipak-pawar) [reply kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/DefaultConfiguration.java]: With `\\s*,\\s*` regex following cases will be covered:
* mykubernetes.json 
* mykubernetes.json, mysecondkubernetes.json
* mykubernetes.json , mysecondkubernetes.json

Comment (spisiakm) [reply kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/DefaultConfiguration.java]: Well, you're using `String.split` java method for separating multiple values by regex, so if there's just one single value, nothing happens and this value will be returned as is.

**Comment by lordofthejars** [comment]
Now looks good to me, if @dipak-pawar agrees, then tomorrow I'll release Cube 2.0.0

**Comment by lordofthejars** [comment]
Also @spisiakm can you fix the build? There are some test failing because of the change (`org.arquillian.cube.kubernetes.reporter.TakeKubernetesResourcesInformationTest`) 

Thanks.

**Comment by spisiakm** [comment]
I'll fix that tomorrow morning.

**Comment by dipak-pawar** [comment]
Thank you @spisiakm. Lgtm

**Pull request merged**